File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
      priority_queue<int, vector<int>, greater<int>> pq;
        int i=0, j=0;
        for(i=0, j=0; i < nums1.size() && j < nums2.size(); i++, j++){
            pq.push(nums1[i]);
            pq.push(nums2[j]);
        }
        if(i < nums1.size()){
            while(i != nums1.size()){
                pq.push(nums1[i]);
                i++;    
            }
        }
        
        if(j < nums2.size()){
            while(j != nums2.size()){
                pq.push(nums2[j]);
                j++;    
            }
        }
        
        if(pq.size()%2 != 0){
            int s = pq.size()/2;
            while(s--){
                pq.pop();
            }
            return pq.top();
        }
        else{
            int prev = 0;
            int s = pq.size()/2;
            while(s--){
                prev = pq.top();
                pq.pop();
            }
            cout<<prev<<" "<<pq.top();
            return (double)(prev + pq.top())/2;
        }  
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1, nums2) -> float:
        len1 = len(nums1)
        len2 = len(nums2)
        #swap lens if len1 > len2 
        if (len1 > len2): 
            return self.findMedianSortedArrays(nums2, nums1)
        low = 0
        high = len1
        left = (len1 + len2 + 1) // 2
        n = len1 + len2
        while low <= high:
            mid1 = (low + high ) // 2
            mid2 = left - mid1
            l1 = l2 = float('-inf')
            r1 = r2 = float('inf')

            if mid1 < len1:
                r1 = nums1[mid1] 
            if mid2 < len2:
                r2 = nums2[mid2] 
            if mid1 - 1 >= 0:
                l1 = nums1[mid1 - 1]
            if mid2 - 1 >= 0:
                l2 = nums2[mid2 - 1]

            if l1 <= r2 and l2 <= r1:
                if n%2 == 1:
                    return max(l1, l2)
                else:
                    return (max(l1, l2) + min(r1, r2)) / 2.0
            elif l1 > r2:
                high = mid1 - 1
            else:
                low = mid1 + 1	
        return 0	
        
			
